packages/contracts-bedrock/test/LivenessGuard.t.sol

```diff
@@ -0,0 +1,318 @@
+        safeInstance.removeOwner({ prevOwner: address(0), owner: ownerToRemove, threshold: 1 });
+        assertEq(livenessGuard.ownersBeforeLength(), 0);
+
+        assertFalse(safeInstance.safe.isOwner(ownerToRemove));
+        assertEq(livenessGuard.lastLive(ownerToRemove), 0);
+    }
+
+    /// @dev Tests that the guard correctly adds an owner to the lastLive mapping when it is added
+    function test_addOwner_succeeds() external {
+        address ownerToAdd = makeAddr("new owner");
+        assertEq(livenessGuard.lastLive(ownerToAdd), 0);
+        assertFalse(safeInstance.safe.isOwner(ownerToAdd));
+
+        assertEq(livenessGuard.ownersBeforeLength(), 0);
+        safeInstance.addOwnerWithThreshold({ owner: ownerToAdd, threshold: 1 });
+        assertEq(livenessGuard.ownersBeforeLength(), 0);
+
+        assertTrue(safeInstance.safe.isOwner(ownerToAdd));
+        assertEq(livenessGuard.lastLive(ownerToAdd), block.timestamp);
+    }
+
+    /// @dev Tests that the guard correctly adds an owner to the lastLive mapping when it is added
+    function test_swapOwner_succeeds() external {
+        address ownerToRemove = safeInstance.owners[0];
+        assertGe(livenessGuard.lastLive(ownerToRemove), 0);
+        assertTrue(safeInstance.safe.isOwner(ownerToRemove));
+
+        address ownerToAdd = makeAddr("new owner");
+        assertEq(livenessGuard.lastLive(ownerToAdd), 0);
+        assertFalse(safeInstance.safe.isOwner(ownerToAdd));
+
+        assertEq(livenessGuard.ownersBeforeLength(), 0);
+        safeInstance.swapOwner({ prevOwner: address(0), oldOwner: ownerToRemove, newOwner: ownerToAdd });
+        assertEq(livenessGuard.ownersBeforeLength(), 0);
+
+        assertFalse(safeInstance.safe.isOwner(ownerToRemove));
+        assertEq(livenessGuard.lastLive(ownerToRemove), 0);
+
+        assertTrue(safeInstance.safe.isOwner(ownerToAdd));
+        assertEq(livenessGuard.lastLive(ownerToAdd), block.timestamp);
+    }
+}
+
+contract LivenessGuard_FuzzOwnerManagement_Test is StdCheats, StdUtils, LivenessGuard_TestInit {
+    using SafeTestLib for SafeInstance;
+
+    /// @dev Enumerates the possible owner management operations
+    enum OwnerOp {
+        Add,
+        Remove,
+        Swap
+    }
+
+    /// @dev Describes a change to be made to the safe
+    struct OwnerChange {
+        uint8 operation; // used to choose an OwnerOp
+        uint256 newThreshold;
+        uint256 ownerIndex;
+    }
+
+    /// @dev Maps addresses to private keys
+    mapping(address => uint256) privateKeys;
+
+    /// @dev Tests that the guard correctly manages the lastLive mapping when owners are added, removed, or swapped
+    function testFuzz_OwnerManagement_works(
+        uint256 initialOwners,
+        uint256 threshold,
+        OwnerChange[] memory changes
+    )
+        external
+    {
+        vm.assume(changes.length < 20);
+        // Initialize the safe with more owners than changes, to ensure we don't try to remove them all
```

Comment:
What do you think about rolling forward the timestamp by a random uint8 each time?

Reply:
I like it! https://github.com/ethereum-optimism/optimism/blob/jm/safe-liveness/packages/contracts-bedrock/test/LivenessGuard.t.sol#L255